param_sweep.py

```diff
@@ -0,0 +1,374 @@
+        'Value(s) for log_2(sigma) of secret s, as inputs to a discrete '
+        'gaussian distribution (unless specified otherwise.) This can be a\n'
+        '* list: looks like -s=6.1,7.2,7.5 or --sigma_s=6,7,8\n'
+        '* range: looks like -s=6:8:0.2 or --sigma-s=6:8 (increments by 1)\n'
+        '* single value: looks like --sigma_s=5\n',
+                        required=True)
+    parser.add_argument('-u', '--uniform_s',
+                        help='Whether to use a uniform distribution for secret '
+                        'key s. If unset, defaults to using a discrete '
+                        'gaussian distribution. Useful for FHE schemes.',
+                        action='store_true')
+    parser.add_argument('-q',
+                        type=str,
+                        help='Value(s) for the modulus q. This can be a\n'
+                        '* list: looks like -q=25,70,80,100\n'
+                        '* range: looks like -q=25:125:25\n'
+                        '* single value: looks like -q=200\n',
+                        default='4294967296', # q=2**32
+                        )
+    parser.add_argument('-f', '--fast',
+                        help=
+        'Use LWE.Estimate.Rough, for a faster estimate. '
+        'If unset, defaults to using LWE.Estimate which is slower.',
+                        action='store_true')
+    parser.add_argument(
+        '--security_cutoff',
+        type=int,
+        help='Generate a second graph, with parameters that give more security '
+        'than the security cutoff labeled with a different value in the graph',
+        default=0,
+        required=False)
+    parser.add_argument(
+        '--output_dir',
+        type=str,
+        help='The directory to output the resulting pickle and graphs',
+        default='')
+    parser.add_argument('--num_proc',
+                        type=int,
+                        help='The number of parallel processes to launch',
+                        default=8)
+    parser.add_argument(
+        '--make_pickle',
+        help=
+        'Whether to dump a pickle file of the intermediate computation results',
+        default=False,
+        action='store_true')
+    parser.add_argument(
+        '--load_pickle',
+        type=str,
+        help='Path to a pickle to load. Overrides any computation',
+        default='')
+    args = parser.parse_args()
+    n = parse_arg(args.n)
+    q = parse_arg(args.q)
+    sigma_e = parse_arg(args.sigma_e)
+    sigma_s = parse_arg(args.sigma_s)
+
+    log_level = 0
+    output_dir = args.output_dir
+    if not output_dir:
+        output_dir = os.path.dirname(os.path.realpath(__file__))
+    assert args.num_proc >= 1, 'need at least one process to execute'
+    file_name = time.strftime('%d-%m-%Y_%H-%M-%S')
+    file_name = os.path.join(output_dir, file_name)
+
+    if not args.load_pickle:
+        result_dict = multiprocessing.Manager().dict()
+        work = []
+        for n_ in n:
```

Comment:
I think this should be a standalone function: take the Cartesian product of these parameter ranges and produce a dictionary of the results, possibly in parallel. This could be useful in other contexts too.

Reply:
Sounds good. I put the functionality of doing a parameter sweep (without graphing) into a separate function, `parameter_sweep`.